apiserver/facades/client/machinemanager/instanceconfig_test.go

```diff
@@ -4,93 +4,80 @@
 
-	c.Check(instanceConfig.APIInfo.Addrs, gc.DeepEquals, apiAddrs)
-	toolsURL := fmt.Sprintf("https://%s/model/%s/tools/%s",
-		apiAddrs[0], jujutesting.ModelTag.Id(), instanceConfig.AgentVersion())
-	c.Assert(instanceConfig.ToolsList().URLs(), jc.DeepEquals, map[version.Binary][]string{
-		instanceConfig.AgentVersion(): {toolsURL},
-	})
+	return ctrl
 }
 
-func (s *machineConfigSuite) TestMachineConfigNoArch(c *gc.C) {
-	apiParams := params.AddMachineParams{
-		Jobs:       []model.MachineJob{model.JobHostUnits},
-		InstanceId: instance.Id("1234"),
-		Nonce:      "foo",
-	}
-	machines, err := apiclient.NewClient(s.APIState).AddMachines([]params.AddMachineParams{apiParams})
-	c.Assert(err, jc.ErrorIsNil)
-	c.Assert(len(machines), gc.Equals, 1)
-	systemState, err := s.StatePool.SystemState()
-	c.Assert(err, jc.ErrorIsNil)
-	_, err = machinemanager.InstanceConfig(systemState, machinemanager.StateBackend(s.State), machines[0].Machine, apiParams.Nonce, "")
-	c.Assert(err, gc.ErrorMatches, fmt.Sprintf("arch is not set for %q", "machine-"+machines[0].Machine))
-}
+func (s *machineConfigSuite) TestMachineConfig(c *gc.C) {
+	ctrl := s.setup(c)
+	defer ctrl.Finish()
 
-func (s *machineConfigSuite) TestMachineConfigNoTools(c *gc.C) {
-	s.PatchValue(&envtools.DefaultBaseURL, "")
+	s.st.EXPECT().Model().Return(s.model, nil)
+	s.model.EXPECT().Config().Return(config.New(config.UseDefaults, dummy.SampleConfig().Merge(coretesting.Attrs{
+		"agent-version":            "2.6.6",
+		"enable-os-upgrade":        true,
+		"enable-os-refresh-update": true,
+	})))
+
+	machine0 := mocks.NewMockMachine(ctrl)
+	machine0.EXPECT().Series().Return("focal").AnyTimes()
+	machine0.EXPECT().Tag().Return(names.NewMachineTag("0")).AnyTimes()
 	hc := instance.MustParseHardware("mem=4G arch=amd64")
-	apiParams := params.AddMachineParams{
-		Series:                  "quantal",
-		Jobs:                    []model.MachineJob{model.JobHostUnits},
-		InstanceId:              instance.Id("1234"),
-		Nonce:                   "foo",
-		HardwareCharacteristics: hc,
-		Addrs:                   params.FromProviderAddresses(network.NewMachineAddress("1.2.3.4").AsProviderAddress()),
-	}
-	machines, err := apiclient.NewClient(s.APIState).AddMachines([]params.AddMachineParams{apiParams})
-	c.Assert(err, jc.ErrorIsNil)
-	systemState, err := s.StatePool.SystemState()
+	machine0.EXPECT().HardwareCharacteristics().Return(&hc, nil)
+	machine0.EXPECT().SetPassword(gomock.Any()).Return(nil)
+	s.st.EXPECT().Machine("0").Return(machine0, nil)
+
+	storageCloser := mocks.NewMockStorageCloser(ctrl)
+	storageCloser.EXPECT().AllMetadata().Return([]binarystorage.Metadata{{
+		Version: "2.6.6-ubuntu-amd64",
+	}}, nil)
+	storageCloser.EXPECT().Close().Return(nil)
+	s.st.EXPECT().ToolsStorage().Return(storageCloser, nil)
+
+	s.ctrlSt.EXPECT().APIHostPortsForAgents().Return([]network.SpaceHostPorts{{{
+		SpaceAddress: network.NewSpaceAddress("0.2.4.6", network.WithScope(network.ScopeCloudLocal)),
```

Comment:
There is a suggestion to use valid IPs in tests

The block 0.0.0.0/8 , which is to say everything from 0.0.0.0 to 0.255.255.255 is reserved for “this host on this network.”

Reply:
I could have sworn I copied this over, but looks like I copied 0.2.4.6 instead of 1.2.3.4